### ISSUE TITLE
Training language models to follow instructions with human feedback

### ISSUE THREAD

**Issue by ooooohira** [ooooohira/paper_summary#3]
### 論文リンク
[https://arxiv.org/abs/2203.02155](https://arxiv.org/abs/2203.02155)

### 著者
Long Ouyang, Jeff Wu, Xu Jiang, Diogo Almeida, Carroll L. Wainwright, Pamela Mishkin, Chong Zhang, Sandhini Agarwal, Katarina Slama, Alex Ray, John Schulman, Jacob Hilton, Fraser Kelton, Luke Miller, Maddie Simens, Amanda Askell, Peter Welinder, Paul Christiano, Jan Leike, Ryan Lowe

### 投稿年
Fri, 4 Mar 2022

## 新規性や利点
強化学習の枠組みを用いて、人間の嗜好を最適化するモデルを学習することで、要約の質を大幅に改善することが可能であることを示す。

## 提案手法
<img width="991" alt="スクリーンショット 2024-04-21 11 32 27" src="https://github.com/ooooohira/paper_summary/assets/131585401/979ae19a-7c84-459f-8db0-7cf7967b2db7">

1 Collect human feedback: Reddit 投稿ごとに、現在の方策、初期の方針、Reddit TL;DR dataset の参考要約、そして様々なベースライン(おそらく教師あり学習でfine-tuningされたモデルの出力)など、いくつかのソースから要約を抽出し、それらをペアにしたものをまとめて人間評価者に送り、与えられた Reddit 投稿に対してより良い要約を選ぶように依頼する。
この時、教師あり学習でfine-tuningされたモデルの出力と、元々のReddit TL;DR datasetに含まれていた参考要約がその都度どれくらい選ばれたのかの監視して、評価者の好みが変わっていないか監視したり、いくつかのペアをまとめて評価者に送って評価させるようにした。
2 Train reward model: 報酬モデル(GPT-3ベースのfine-tuningずみのモデル)を学習。

3 Train policy with PPO: PPOアルゴリズムをもとに方策モデル(GPT-3ベースのfine-tuningずみのモデル)を学習する。また、この時に使用する価値関数モデルもGPT-3ベースのfine-tuningずみのモデルとなっている。

## 実験

1. human feedback model (本研究での手法)と、fine-tuning のみと、pretrain のみでさまざまなモデルサイズを用いて性能を比較した。
2. CNN/DMデータセットを用いてゼロショット転移の性能を測った。

## 結果と考察

1. 性能評価
<img width="877" alt="スクリーンショット 2024-04-21 13 54 17" src="https://github.com/ooooohira/paper_summary/assets/131585401/46e975ac-5b17-40dd-8618-2455d26e115b">

人間が作成した要約(Reference Summaries)と比べて、人間がどれくらい良いかを選んだかの割合。

<p align="center">
<img width="350" alt="スクリーンショット 2024-04-21 13 55 26" src="https://github.com/ooooohira/paper_summary/assets/131585401/7f66ee69-9efb-4b6e-a6db-4877d8103482">
</p>

全体的な品質 (Overall): これらの要素を総合した全体的な品質
網羅性 (Coverage): 元の投稿の重要な情報がどれほどカバーされているか
コヒーレンス (Coherence): 要約単体でどれほど読みやすいか
正確性 (Accuracy): 要約に記載されている情報が元の投稿の内容とどれほど一致しているか
これら4つの観点で人間が評価した結果。

2. ゼロショット転移
また、CNN/DMデータセットの要約をゼロショットで高い精度で転移可能なことも判明した。

## 感想や疑問点
1. どうやって3つのステップを同期させて処理しているのかの詳細。
2. どれくらいの量のデータを人間に比較してもらったのか。

## これからやるべきこと
TPOアルゴリズムの理解

## 参考
